<!doctype html>
<html class="no-js" lang="en" data-content_root="./">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="genindex.html" /><link rel="search" title="Search" href="search.html" /><link rel="prev" title="Changelog" href="news.html" />
        <link rel="canonical" href="https://clustering-benchmarks.gagolewski.com/z_bibliography.html" />

    <link rel="shortcut icon" href="https://www.gagolewski.com/_static/img/clustbench.png"/><!-- Generated with Sphinx 8.2.3 and Furo 2024.08.06 -->
        <title>References - Clustering Benchmarks</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=b86133f3" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo.css?v=354aac6f" />
    <link rel="stylesheet" type="text/css" href="_static/plot_directive.css" />
    <link rel="stylesheet" type="text/css" href="_static/graphviz.css?v=4ae1632d" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo-extensions.css?v=302659d7" />
    <link rel="stylesheet" type="text/css" href="_static/css/custom.css?v=1981fb80" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --admonition-font-size: 95%;
  --admonition-title-font-size: 95%;
  --color-brand-primary: red;
  --color-brand-content: #CC3333;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  --admonition-font-size: 95%;
  --admonition-title-font-size: 95%;
  --color-brand-primary: #ff2b53;
  --color-brand-content: #dd3333;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  --admonition-font-size: 95%;
  --admonition-title-font-size: 95%;
  --color-brand-primary: #ff2b53;
  --color-brand-content: #dd3333;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-with-moon" viewBox="0 0 24 24">
    <title>Auto light/dark, in light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path style="opacity: 50%" d="M 5.411 14.504 C 5.471 14.504 5.532 14.504 5.591 14.504 C 3.639 16.319 4.383 19.569 6.931 20.352 C 7.693 20.586 8.512 20.551 9.25 20.252 C 8.023 23.207 4.056 23.725 2.11 21.184 C 0.166 18.642 1.702 14.949 4.874 14.536 C 5.051 14.512 5.231 14.5 5.411 14.5 L 5.411 14.504 Z"/>
      <line x1="14.5" y1="3.25" x2="14.5" y2="1.25"/>
      <line x1="14.5" y1="15.85" x2="14.5" y2="17.85"/>
      <line x1="10.044" y1="5.094" x2="8.63" y2="3.68"/>
      <line x1="19" y1="14.05" x2="20.414" y2="15.464"/>
      <line x1="8.2" y1="9.55" x2="6.2" y2="9.55"/>
      <line x1="20.8" y1="9.55" x2="22.8" y2="9.55"/>
      <line x1="10.044" y1="14.006" x2="8.63" y2="15.42"/>
      <line x1="19" y1="5.05" x2="20.414" y2="3.636"/>
      <circle cx="14.5" cy="9.55" r="3.6"/>
    </svg>
  </symbol>
  <symbol id="svg-moon-with-sun" viewBox="0 0 24 24">
    <title>Auto light/dark, in dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path d="M 8.282 7.007 C 8.385 7.007 8.494 7.007 8.595 7.007 C 5.18 10.184 6.481 15.869 10.942 17.24 C 12.275 17.648 13.706 17.589 15 17.066 C 12.851 22.236 5.91 23.143 2.505 18.696 C -0.897 14.249 1.791 7.786 7.342 7.063 C 7.652 7.021 7.965 7 8.282 7 L 8.282 7.007 Z"/>
      <line style="opacity: 50%" x1="18" y1="3.705" x2="18" y2="2.5"/>
      <line style="opacity: 50%" x1="18" y1="11.295" x2="18" y2="12.5"/>
      <line style="opacity: 50%" x1="15.316" y1="4.816" x2="14.464" y2="3.964"/>
      <line style="opacity: 50%" x1="20.711" y1="10.212" x2="21.563" y2="11.063"/>
      <line style="opacity: 50%" x1="14.205" y1="7.5" x2="13.001" y2="7.5"/>
      <line style="opacity: 50%" x1="21.795" y1="7.5" x2="23" y2="7.5"/>
      <line style="opacity: 50%" x1="15.316" y1="10.184" x2="14.464" y2="11.036"/>
      <line style="opacity: 50%" x1="20.711" y1="4.789" x2="21.563" y2="3.937"/>
      <circle style="opacity: 50%" cx="18" cy="7.5" r="2.169"/>
    </svg>
  </symbol>
  <symbol id="svg-pencil" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-pencil-code">
      <path d="M4 20h4l10.5 -10.5a2.828 2.828 0 1 0 -4 -4l-10.5 10.5v4" />
      <path d="M13.5 6.5l4 4" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
  <symbol id="svg-eye" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-eye-code">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0" />
      <path
        d="M11.11 17.958c-3.209 -.307 -5.91 -2.293 -8.11 -5.958c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6c-.21 .352 -.427 .688 -.647 1.008" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="index.html"><div class="brand">Clustering Benchmarks</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto colour theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky">
<div class="sidebar-logo-container">
  <a class="sidebar-brand" href="index.html"><img class="sidebar-logo" src="https://www.gagolewski.com/_static/img/clustbench.png" alt="Logo"/></a>
</div>

<span class="sidebar-brand-text">
<a class="sidebar-brand" href="index.html">Clustering Benchmarks</a>
</span>
<div class="sidebar-brand">
1.1.6
</div>
<form class="sidebar-search-container" method="get" action="search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Methodology</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="index.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/true-vs-predicted.html">True vs Predicted Clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/noise-points.html">Noise Points</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/many-partitions.html">There Can Be Many Valid Partitions</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Benchmark Batteries</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="weave/suite-v1.html">Benchmark Suite (v1.1.0)</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/data-v1.html">Explore Datasets (v1.1.0)</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/results-v1.html">Clustering Results Repository (v1.1.0)</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/file-format.html">File Format Specification</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/how-to-access.html">Access from Python, R, MATLAB, etc.</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/contributing.html">Contributing</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Python API</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="weave/clustbench-usage.html">Using <em>clustbench</em></a></li>
<li class="toctree-l1"><a class="reference internal" href="clustbench-documentation.html">Documentation</a></li>
<li class="toctree-l1"><a class="reference external" href="https://github.com/gagolews/clustering-benchmarks">Source Code (GitHub)</a></li>
<li class="toctree-l1"><a class="reference external" href="https://pypi.org/project/clustering-benchmarks">PyPI Entry</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">See Also</span></p>
<ul>
<li class="toctree-l1"><a class="reference external" href="https://www.gagolewski.com/">Author's Homepage</a></li>
<li class="toctree-l1"><a class="reference external" href="https://datawranglingpy.gagolewski.com/">Data Wrangling in Python</a></li>
<li class="toctree-l1"><a class="reference external" href="https://deepr.gagolewski.com">Deep R Programming</a></li>
<li class="toctree-l1"><a class="reference external" href="https://genieclust.gagolewski.com/">genieclust Package</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Appendix</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="weave/colouriser.html">Colouriser: A Planar Data Editor</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/external-validity-measures.html">External Cluster Validity Measures</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/internal-validity-measures.html">Side Note: Internal <em>(In)Validity</em> Measures</a></li>
<li class="toctree-l1"><a class="reference internal" href="news.html">Changelog</a></li>
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">References</a></li>
</ul>

</div></div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          <div class="edit-this-page">
  <a class="muted-link" href="https://github.com/gagolews/clustering-benchmarks//issues/" title="Edit this page">
    <svg><use href="#svg-pencil"></use></svg>
    <span class="visually-hidden">Edit this page</span>
  </a>
</div><div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto colour theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <section class="tex2jax_ignore mathjax_ignore" id="references">
<h1>References<a class="headerlink" href="#references" title="Link to this heading">¶</a></h1>
<section id="how-to-cite">
<h2>How to Cite<a class="headerlink" href="#how-to-cite" title="Link to this heading">¶</a></h2>
<p>When using this framework in research publications, please
cite <span id="id1">[<a class="reference internal" href="#id6" title="Gagolewski, M. (2022).  A framework for benchmarking clustering algorithms. SoftwareX, 20:101270. URL: https://clustering-benchmarks.gagolewski.com/, DOI: 10.1016/j.softx.2022.101270.">21</a>]</span> as specified below. Thank you.</p>
<p>Additionally, please mention the benchmark suite project
<span id="id2">[<a class="reference internal" href="#id7" title="Gagolewski, M. and others. (2022).  A benchmark suite for clustering algorithms: Version 1.1.0. URL: https://github.com/gagolews/clustering-data-v1/releases/tag/v1.1.0, DOI: 10.5281/zenodo.7088171.">28</a>]</span> as well as the literature references
listed in the description files corresponding to each dataset studied.</p>
</section>
<section id="bibliography">
<h2>Bibliography<a class="headerlink" href="#bibliography" title="Link to this heading">¶</a></h2>
<div>
<div class="docutils container" id="id3">
<div role="list" class="citation-list">
<div class="citation" id="id51" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>1<span class="fn-bracket">]</span></span>
<p>Arbelaitz, O., Gurrutxaga, I., Muguerza, J., Pérez, J.M., and Perona, I. (2013).  An extensive comparative study of cluster validity indices. <em>Pattern Recognition</em>, 46(1):243–256. DOI: <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.patcog.2012.07.021">https://doi.org/10.1016/j.patcog.2012.07.021</a>.</p>
</div>
<div class="citation" id="id42" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>2<span class="fn-bracket">]</span></span>
<p>Ball, G.H. and Hall, D.J. (1965).  ISODATA: A novel method of data analysis and pattern classification. Technical Report AD699616, Stanford Research Institute.</p>
</div>
<div class="citation" id="id75" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>3<span class="fn-bracket">]</span></span>
<p>Bezdek, J.C., Keller, J.M., Krishnapuram, R., Kuncheva, L.I., and Pal, N.R. (1999).  Will the real iris data please stand up? <em>IEEE Transactions on Fuzzy Systems</em>, 7(3):368–369. DOI: <a class="reference external" href="https://doi.org/10.1109/91.771092">10.1109/91.771092</a>.</p>
</div>
<div class="citation" id="id52" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>4<span class="fn-bracket">]</span></span>
<p>Bezdek, J.C. and Pal, N.R. (1998).  Some new indexes of cluster validity. <em>IEEE Transactions on Systems, Man, and Cybernetics, Part B (Cybernetics)</em>, 28(3):301–315. DOI: <a class="reference external" href="https://doi.org/10.1109/3477.678624">10.1109/3477.678624</a>.</p>
</div>
<div class="citation" id="id76" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>5<span class="fn-bracket">]</span></span>
<p>Blum, A., Hopcroft, J., and Kannan, R. (2020).  <em>Foundations of Data Science</em>. Cambridge University Press. URL: <a class="reference external" href="https://www.cs.cornell.edu/jeh/book.pdf">https://www.cs.cornell.edu/jeh/book.pdf</a>.</p>
</div>
<div class="citation" id="id24" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>6<span class="fn-bracket">]</span></span>
<p>Buitinck, L. and others. (2013).  API design for machine learning software: Experiences from the scikit-learn project. In: <em>ECML PKDD Workshop: Languages for Data Mining and Machine Learning</em>, pp. 108–122.</p>
</div>
<div class="citation" id="id39" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>7<span class="fn-bracket">]</span></span>
<p>Caliński, T. and Harabasz, J. (1974).  A dendrite method for cluster analysis. <em>Communications in Statistics</em>, 3(1):1–27. DOI: <a class="reference external" href="https://doi.org/10.1080/03610927408827101">10.1080/03610927408827101</a>.</p>
</div>
<div class="citation" id="id69" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>8<span class="fn-bracket">]</span></span>
<p>Chang, H. and Yeung, D.Y. (2008).  Robust path-based spectral clustering. <em>Pattern Recognition</em>, 41(1):191–203.</p>
</div>
<div class="citation" id="id17" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>9<span class="fn-bracket">]</span></span>
<p>Crouse, D.F. (2016).  On implementing 2D rectangular assignment algorithms. <em>IEEE Transactions on Aerospace and Electronic Systems</em>, 52(4):1679–1696. DOI: <a class="reference external" href="https://doi.org/10.1109/TAES.2016.140952">10.1109/TAES.2016.140952</a>.</p>
</div>
<div class="citation" id="id26" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>10<span class="fn-bracket">]</span></span>
<p>Dasgupta, S. and Ng, V. (2009).  Single data, multiple clusterings. In: <em>Proc. NIPS Workshop Clustering: Science or Art? Towards Principled Approaches</em>. URL: <a class="reference external" href="https://clusteringtheory.org">https://clusteringtheory.org</a>.</p>
</div>
<div class="citation" id="id40" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>11<span class="fn-bracket">]</span></span>
<p>Davies, D.L. and Bouldin, D.W. (1979).  A cluster separation measure. <em>IEEE Transactions on Pattern Analysis and Machine Intelligence</em>, PAMI–1(2):224–227. DOI: <a class="reference external" href="https://doi.org/10.1109/TPAMI.1979.4766909">10.1109/TPAMI.1979.4766909</a>.</p>
</div>
<div class="citation" id="id56" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>12<span class="fn-bracket">]</span></span>
<p>Dua, D. and Graff, C. (2022).  <em>UCI Machine Learning Repository</em>. <a class="reference external" href="http://archive.ics.uci.edu/ml">http://archive.ics.uci.edu/ml</a>.</p>
</div>
<div class="citation" id="id41" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>13<span class="fn-bracket">]</span></span>
<p>Dunn, J.C. (1974).  A fuzzy relative of the ISODATA process and its use in detecting compact well-separated clusters. <em>Journal of Cybernetics</em>, 3(3):32–57. DOI: <a class="reference external" href="https://doi.org/10.1080/01969727308546046">10.1080/01969727308546046</a>.</p>
</div>
<div class="citation" id="id46" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>14<span class="fn-bracket">]</span></span>
<p>Edwards, A.W.F. and Cavalli-Sforza, L.L. (1965).  A method for cluster analysis. <em>Biometrics</em>, 21(2):362–375. DOI: <a class="reference external" href="https://doi.org/10.2307/2528096">10.2307/2528096</a>.</p>
</div>
<div class="citation" id="id66" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>15<span class="fn-bracket">]</span></span>
<p>Fisher, R.A. (1936).  The use of multiple measurements in taxonomic problems. <em>Annals of Eugenics</em>, 7:179–188.</p>
</div>
<div class="citation" id="id60" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>16<span class="fn-bracket">]</span></span>
<p>Fowlkes, E.B. and Mallows, C.L. (1983).  A method for comparing two hierarchical clusterings. <em>Journal of the American Statistical Association</em>, 78(383):553–569.</p>
</div>
<div class="citation" id="id27" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>17<span class="fn-bracket">]</span></span>
<p>Fränti, P. and Sieranoja, S. (2018).  K-means properties on six clustering benchmark datasets. <em>Applied Intelligence</em>, 48(12):4743–4759. DOI: <a class="reference external" href="https://doi.org/10.1007/s10489-018-1238-7">10.1007/s10489-018-1238-7</a>.</p>
</div>
<div class="citation" id="id67" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>18<span class="fn-bracket">]</span></span>
<p>Fränti, P. and Virmajoki, O. (2006).  Iterative shrinking method for clustering problems. <em>Pattern Recognition</em>, 39(5):761–765.</p>
</div>
<div class="citation" id="id74" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>19<span class="fn-bracket">]</span></span>
<p>Fu, L. and Medico, E. (2007).  FLAME: A novel fuzzy clustering method for the analysis of DNA microarray data. <em>BMC bioinformatics</em>, 8:3.</p>
</div>
<div class="citation" id="id11" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>20<span class="fn-bracket">]</span></span>
<p>Gagolewski, M. (2021).  genieclust: Fast and robust hierarchical clustering. <em>SoftwareX</em>, 15:100722. URL: <a class="reference external" href="https://genieclust.gagolewski.com/">https://genieclust.gagolewski.com/</a>, DOI: <a class="reference external" href="https://doi.org/10.1016/j.softx.2021.100722">10.1016/j.softx.2021.100722</a>.</p>
</div>
<div class="citation" id="id6" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id1">21</a><span class="fn-bracket">]</span></span>
<p>Gagolewski, M. (2022).  A framework for benchmarking clustering algorithms. <em>SoftwareX</em>, 20:101270. URL: <a class="reference external" href="https://clustering-benchmarks.gagolewski.com/">https://clustering-benchmarks.gagolewski.com/</a>, DOI: <a class="reference external" href="https://doi.org/10.1016/j.softx.2022.101270">10.1016/j.softx.2022.101270</a>.</p>
</div>
<div class="citation" id="id9" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>22<span class="fn-bracket">]</span></span>
<p>Gagolewski, M. (2025).  <em>Deep R Programming</em>. Zenodo, Melbourne. ISBN 978-0-6455719-2-9. URL: <a class="reference external" href="https://deepr.gagolewski.com/">https://deepr.gagolewski.com/</a>, DOI: <a class="reference external" href="https://doi.org/10.5281/zenodo.7490464">10.5281/zenodo.7490464</a>.</p>
</div>
<div class="citation" id="id8" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>23<span class="fn-bracket">]</span></span>
<p>Gagolewski, M. (2025).  <em>Minimalist Data Wrangling with Python</em>. Zenodo, Melbourne. ISBN 978-0-6455719-1-2. URL: <a class="reference external" href="https://datawranglingpy.gagolewski.com/">https://datawranglingpy.gagolewski.com/</a>, DOI: <a class="reference external" href="https://doi.org/10.5281/zenodo.6451068">10.5281/zenodo.6451068</a>.</p>
</div>
<div class="citation" id="id5" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>24<span class="fn-bracket">]</span></span>
<p>Gagolewski, M. (2025).  Normalised clustering accuracy: An asymmetric external cluster validity measure. <em>Journal of Classification</em>, 42:2–30. URL: <a class="reference external" href="https://link.springer.com/content/pdf/10.1007/s00357-024-09482-2.pdf">https://link.springer.com/content/pdf/10.1007/s00357-024-09482-2.pdf</a>, DOI: <a class="reference external" href="https://doi.org/10.1007/s00357-024-09482-2">10.1007/s00357-024-09482-2</a>.</p>
</div>
<div class="citation" id="id12" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>25<span class="fn-bracket">]</span></span>
<p>Gagolewski, M., Bartoszuk, M., and Cena, A. (2016).  Genie: A new, fast, and outlier-resistant hierarchical clustering algorithm. <em>Information Sciences</em>, 363:8–23. URL: <a class="reference external" href="https://arxiv.org/pdf/2209.05757">https://arxiv.org/pdf/2209.05757</a>, DOI: <a class="reference external" href="https://doi.org/10.1016/j.ins.2016.05.003">10.1016/j.ins.2016.05.003</a>.</p>
</div>
<div class="citation" id="id10" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>26<span class="fn-bracket">]</span></span>
<p>Gagolewski, M., Bartoszuk, M., and Cena, A. (2021).  Are cluster validity measures (in)valid? <em>Information Sciences</em>, 581:620–636. URL: <a class="reference external" href="https://arxiv.org/pdf/2208.01261">https://arxiv.org/pdf/2208.01261</a>, DOI: <a class="reference external" href="https://doi.org/10.1016/j.ins.2021.10.004">10.1016/j.ins.2021.10.004</a>.</p>
</div>
<div class="citation" id="id4" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>27<span class="fn-bracket">]</span></span>
<p>Gagolewski, M., Cena, A., Bartoszuk, M., and Brzozowski, L. (2025).  Clustering with minimum spanning trees: How good can it be? <em>Journal of Classification</em>, 42:90–112. URL: <a class="reference external" href="https://link.springer.com/content/pdf/10.1007/s00357-024-09483-1.pdf">https://link.springer.com/content/pdf/10.1007/s00357-024-09483-1.pdf</a>, DOI: <a class="reference external" href="https://doi.org/10.1007/s00357-024-09483-1">10.1007/s00357-024-09483-1</a>.</p>
</div>
<div class="citation" id="id7" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id2">28</a><span class="fn-bracket">]</span></span>
<p>Gagolewski, M. and others. (2022).  <em>A benchmark suite for clustering algorithms: Version 1.1.0</em>. URL: <a class="reference external" href="https://github.com/gagolews/clustering-data-v1/releases/tag/v1.1.0">https://github.com/gagolews/clustering-data-v1/releases/tag/v1.1.0</a>, DOI: <a class="reference external" href="https://doi.org/10.5281/zenodo.7088171">10.5281/zenodo.7088171</a>.</p>
</div>
<div class="citation" id="id72" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>29<span class="fn-bracket">]</span></span>
<p>Gionis, A., Mannila, H., and Tsaparas, P. (2007).  Clustering aggregation. <em>ACM Transactions on Knowledge Discovery from Data</em>, 1(1):4.</p>
</div>
<div class="citation" id="id32" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>30<span class="fn-bracket">]</span></span>
<p>Graves, D. and Pedrycz, W. (2010).  Kernel-based fuzzy clustering and fuzzy clustering: A comparative experimental study. <em>Fuzzy Sets and Systems</em>, 161:522–543. DOI: <a class="reference external" href="https://doi.org/10.1016/j.fss.2009.10.021">10.1016/j.fss.2009.10.021</a>.</p>
</div>
<div class="citation" id="id38" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>31<span class="fn-bracket">]</span></span>
<p>Halkidi, M., Batistakis, Y., and Vazirgiannis, M. (2001).  On clustering validation techniques. <em>Journal of Intelligent Information Systems</em>, pages 107–145. DOI: <a class="reference external" href="https://doi.org/10.1023/A:1012801612483">10.1023/A:1012801612483</a>.</p>
</div>
<div class="citation" id="id14" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>32<span class="fn-bracket">]</span></span>
<p>Hubert, L. and Arabie, P. (1985).  Comparing partitions. <em>Journal of Classification</em>, 2(1):193–218. DOI: <a class="reference external" href="https://doi.org/10.1007/BF01908075">10.1007/BF01908075</a>.</p>
</div>
<div class="citation" id="id73" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>33<span class="fn-bracket">]</span></span>
<p>Jain, A. and Law, M. (2005).  Data clustering: A user's dilemma. <em>Lecture Notes in Computer Science</em>, 3776:1–10.</p>
</div>
<div class="citation" id="id78" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>34<span class="fn-bracket">]</span></span>
<p>Jamil, M., Yang, X.-S., and Zepernick, H.-J. (2013).  8-test functions for global optimization: A comprehensive survey. In: <em>Swarm Intelligence and Bio-Inspired Computation</em>, pp. 193–222. DOI: <a class="reference external" href="https://doi.org/10.1016/B978-0-12-405163-8.00008-9">10.1016/B978-0-12-405163-8.00008-9</a>.</p>
</div>
<div class="citation" id="id34" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>35<span class="fn-bracket">]</span></span>
<p>Karypis, G., Han, E.H., and Kumar, V. (1999).  CHAMELEON: Hierarchical clustering using dynamic modeling. <em>Computer</em>, 32(8):68–75. DOI: <a class="reference external" href="https://doi.org/10.1109/2.781637">10.1109/2.781637</a>.</p>
</div>
<div class="citation" id="id19" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>36<span class="fn-bracket">]</span></span>
<p>Kvalseth, T.O. (1987).  Entropy and correlation: Some comments. <em>IEEE Trans. Systems, Man and Cybernetics</em>, 17(3):517–519. DOI: <a class="reference external" href="https://doi.org/10.1109/TSMC.1987.4309069">10.1109/TSMC.1987.4309069</a>.</p>
</div>
<div class="citation" id="id71" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>37<span class="fn-bracket">]</span></span>
<p>Kärkkäinen, I. and Fränti, P. (2002).  Dynamic local search algorithm for the clustering problem. In: <em>Proc. 16th Intl. Conf. Pattern Recognition'02</em>, volume 2, pp. 240–243. IEEE.</p>
</div>
<div class="citation" id="id49" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>38<span class="fn-bracket">]</span></span>
<p>Lloyd, S.P. (1957).  Least squares quantization in PCM. <em>IEEE Transactions on Information Theory</em>, 28:128–137. Originally a 1957 Bell Telephone Laboratories Research Report; republished in 1982. DOI: <a class="reference external" href="https://doi.org/10.1109/TIT.1982.1056489">10.1109/TIT.1982.1056489</a>.</p>
</div>
<div class="citation" id="id37" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>39<span class="fn-bracket">]</span></span>
<p>Maulik, U. and Bandyopadhyay, S. (2002).  Performance evaluation of some clustering algorithms and validity indices. <em>IEEE Transactions on Pattern Analysis and Machine Intelligence</em>, 24(12):1650–1654. DOI: <a class="reference external" href="https://doi.org/10.1109/TPAMI.2002.1114856">10.1109/TPAMI.2002.1114856</a>.</p>
</div>
<div class="citation" id="id35" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>40<span class="fn-bracket">]</span></span>
<p>McInnes, L., Healy, J., and Astels, S. (2017).  hdbscan: Hierarchical density based clustering. <em>The Journal of Open Source Software</em>, 2(11):205. DOI: <a class="reference external" href="https://doi.org/10.21105/joss.00205">10.21105/joss.00205</a>.</p>
</div>
<div class="citation" id="id57" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>41<span class="fn-bracket">]</span></span>
<p>Meilă, M. and Heckerman, D. (2001).  An experimental comparison of model-based clustering methods. <em>Machine Learning</em>, 42:9–29. DOI: <a class="reference external" href="https://doi.org/10.1023/A:1007648401407">10.1023/A:1007648401407</a>.</p>
</div>
<div class="citation" id="id36" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>42<span class="fn-bracket">]</span></span>
<p>Milligan, G.W. and Cooper, M.C. (1985).  An examination of procedures for determining the number of clusters in a data set. <em>Psychometrika</em>, 50(2):159–179.</p>
</div>
<div class="citation" id="id77" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>43<span class="fn-bracket">]</span></span>
<p>Müller, A.C., Nowozin, S., and Lampert, C.H. (2012).  Information theoretic clustering using minimum spanning trees. In: <em>Proc. German Conference on Pattern Recognition</em>. URL: <a class="reference external" href="https://github.com/amueller/information-theoretic-mst">https://github.com/amueller/information-theoretic-mst</a>.</p>
</div>
<div class="citation" id="id15" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>44<span class="fn-bracket">]</span></span>
<p>Müllner, D. (2013).  fastcluster: Fast hierarchical, agglomerative clustering routines for R and Python. <em>Journal of Statistical Software</em>, 53(9):1–18. DOI: <a class="reference external" href="https://doi.org/10.18637/jss.v053.i09">10.18637/jss.v053.i09</a>.</p>
</div>
<div class="citation" id="id25" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>45<span class="fn-bracket">]</span></span>
<p>Pedregosa, F. and others. (2011).  Scikit-learn: Machine learning in Python. <em>Journal of Machine Learning Research</em>, 12(85):2825–2830. URL: <a class="reference external" href="http://jmlr.org/papers/v12/pedregosa11a.html">http://jmlr.org/papers/v12/pedregosa11a.html</a>.</p>
</div>
<div class="citation" id="id61" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>46<span class="fn-bracket">]</span></span>
<p>Rand, W.M. (1971).  Objective criteria for the evaluation of clustering methods. <em>Journal of the American Statistical Association</em>, 66(336):846–850. DOI: <a class="reference external" href="https://doi.org/10.2307/2284239">10.2307/2284239</a>.</p>
</div>
<div class="citation" id="id18" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>47<span class="fn-bracket">]</span></span>
<p>Rezaei, M. and Fränti, P. (2016).  Set matching measures for external cluster validity. <em>IEEE Transactions on Knowledge and Data Engineering</em>, 28(8):2173–2186. DOI: <a class="reference external" href="https://doi.org/10.1109/TKDE.2016.2551240">10.1109/TKDE.2016.2551240</a>.</p>
</div>
<div class="citation" id="id43" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>48<span class="fn-bracket">]</span></span>
<p>Rousseeuw, P.J. (1987).  Silhouettes: A graphical aid to the interpretation and validation of cluster analysis. <em>Journal of Computational and Applied Mathematics</em>, 20:53–65. DOI: <a class="reference external" href="https://doi.org/10.1016/0377-0427(87)90125-7">10.1016/0377-0427(87)90125-7</a>.</p>
</div>
<div class="citation" id="id64" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>49<span class="fn-bracket">]</span></span>
<p>Sieranoja, S. and Fränti, P. (2019).  Fast and general density peaks clustering. <em>Pattern Recognition Letters</em>, 128:551–558. DOI: <a class="reference external" href="https://doi.org/10.1016/j.patrec.2019.10.019">10.1016/j.patrec.2019.10.019</a>.</p>
</div>
<div class="citation" id="id58" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>50<span class="fn-bracket">]</span></span>
<p>Steinley, D. (2004).  Properties of the Hubert–Arabie adjusted Rand index. <em>Psychological Methods</em>, 9(3):386–396. DOI: <a class="reference external" href="https://doi.org/10.1037/1082-989X.9.3.386">10.1037/1082-989X.9.3.386</a>.</p>
</div>
<div class="citation" id="id31" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>51<span class="fn-bracket">]</span></span>
<p>Thrun, M.C. and Stier, Q. (2021).  Fundamental clustering algorithms suite. <em>SoftwareX</em>, 13:100642. DOI: <a class="reference external" href="https://doi.org/10.1016/j.softx.2020.100642">10.1016/j.softx.2020.100642</a>.</p>
</div>
<div class="citation" id="id30" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>52<span class="fn-bracket">]</span></span>
<p>Thrun, M.C. and Ultsch, A. (2020).  Clustering benchmark datasets exploiting the fundamental clustering problems. <em>Data in Brief</em>, 30:105501. DOI: <a class="reference external" href="https://doi.org/10.1016/j.dib.2020.105501">10.1016/j.dib.2020.105501</a>.</p>
</div>
<div class="citation" id="id84" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>53<span class="fn-bracket">]</span></span>
<p>Ullmann, T., Beer, A., Hünemörder, M., Seidl, T., and Boulesteix, A.-L. (2022).  Over-optimistic evaluation and reporting of novel cluster algorithms: An illustrative study. <em>Advances in Data Analysis and Classification</em>. DOI: <a class="reference external" href="https://doi.org/10.1007/s11634-022-00496-5">10.1007/s11634-022-00496-5</a>.</p>
</div>
<div class="citation" id="id86" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>54<span class="fn-bracket">]</span></span>
<p>Ullmann, T., Hennig, C., and Boulesteix, A.-L. (2022).  Validation of cluster analysis results on validation data: A systematic framework. <em>Wiley Interdisciplinary Reviews: Data Mining and Knowledge Discovery</em>, 12(3):e14444. DOI: <a class="reference external" href="https://doi.org/10.1002/widm.1444">10.1002/widm.1444</a>.</p>
</div>
<div class="citation" id="id29" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>55<span class="fn-bracket">]</span></span>
<p>Ultsch, A. (2005).  Clustering with SOM: U*C. In: <em>Workshop on Self-Organizing Maps</em>, pp. 75–82.</p>
</div>
<div class="citation" id="id70" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>56<span class="fn-bracket">]</span></span>
<p>Veenman, C.J., Reinders, M.J.T., and Backer, E. (2002).  A maximum variance cluster algorithm. <em>IEEE Transactions on Pattern Analysis and Machine Intelligence</em>, 24(9):1273–1280.</p>
</div>
<div class="citation" id="id20" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>57<span class="fn-bracket">]</span></span>
<p>Vinh, N.X., Epps, J., and Bailey, J. (2010).  Information theoretic measures for clusterings comparison: Variants, properties, normalization and correction for chance. <em>Journal of Machine Learning Research</em>, 11(95):2837–2854. URL: <a class="reference external" href="http://jmlr.org/papers/v11/vinh10a.html">http://jmlr.org/papers/v11/vinh10a.html</a>.</p>
</div>
<div class="citation" id="id53" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>58<span class="fn-bracket">]</span></span>
<p>von Luxburg, U., Williamson, R.C., and Guyon, I. (2012).  Clustering: science or art? In: Guyon, I. and others, editors, <em>Proc. ICML Workshop on Unsupervised and Transfer Learning</em>, volume 27 of Proc. Machine Learning Research, pp. 65–79.</p>
</div>
<div class="citation" id="id47" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>59<span class="fn-bracket">]</span></span>
<p>Ward Jr., J.H. (1963).  Hierarchical grouping to optimize an objective function. <em>Journal of the American Statistical Association</em>, 58(301):236–244. DOI: <a class="reference external" href="https://doi.org/10.1080/01621459.1963.10500845">10.1080/01621459.1963.10500845</a>.</p>
</div>
<div class="citation" id="id79" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>60<span class="fn-bracket">]</span></span>
<p>Weise, T. and others. (2014).  Benchmarking optimization algorithms: An open source framework for the traveling salesman problem. <em>IEEE Computational Intelligence Magazine</em>, 9(3):40–52. DOI: <a class="reference external" href="https://doi.org/10.1109/MCI.2014.2326101">10.1109/MCI.2014.2326101</a>.</p>
</div>
<div class="citation" id="id63" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>61<span class="fn-bracket">]</span></span>
<p>Xiao, H., Rasul, K., and Vollgraf, R. (2017).  <em>Fashion-MNIST: A Novel Image Dataset for Benchmarking Machine Learning Algorithms</em>. URL: <a class="reference external" href="https://arxiv.org/pdf/1708.07747.pdf">https://arxiv.org/pdf/1708.07747.pdf</a>, DOI: <a class="reference external" href="https://doi.org/10.48550/arXiv.1708.07747">10.48550/arXiv.1708.07747</a>.</p>
</div>
<div class="citation" id="id50" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>62<span class="fn-bracket">]</span></span>
<p>Xu, Q., Zhang, Q., Liu, J., and Luo, B. (2020).  Efficient synthetical clustering validity indexes for hierarchical clustering. <em>Expert Systems with Applications</em>, 151:113367. DOI: <a class="reference external" href="https://doi.org/10.1016/j.eswa.2020.113367">10.1016/j.eswa.2020.113367</a>.</p>
</div>
<div class="citation" id="id68" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>63<span class="fn-bracket">]</span></span>
<p>Zahn, C.T. (1971).  Graph-theoretical methods for detecting and describing gestalt clusters. <em>IEEE Transactions on Computers</em>, C-20(1):68–86.</p>
</div>
<div class="citation" id="id65" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>64<span class="fn-bracket">]</span></span>
<p>Zhang, T., Ramakrishnan, R., and Livny, M. (1997).  BIRCH: A new data clustering algorithm and its applications. <em>Data Mining and Knowledge Discovery</em>, 1:141–182. DOI: <a class="reference external" href="https://doi.org/10.1023/A:1009783824328">10.1023/A:1009783824328</a>.</p>
</div>
<div class="citation" id="id85" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>65<span class="fn-bracket">]</span></span>
<p>van Mechelen, I., Boulesteix, A.-L., Dangl, R., and others. (2023).  <em>A white paper on good research practices in benchmarking: The case of cluster analysis</em>. DOI: <a class="reference external" href="https://doi.org/10.1002/widm.1511">10.1002/widm.1511</a>.</p>
</div>
</div>
</div>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          
          <a class="prev-page" href="news.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">Changelog</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
              
              
              Copyright &#169; 2020–2025 by <a href="https://www.gagolewski.com">Marek Gagolewski</a>.
              Some rights reserved. Licensed under <a href='https://creativecommons.org/licenses/by-nc-nd/4.0/'>CC BY-NC-ND 4.0</a>.
              Built with <a href="https://sphinx-doc.org/">Sphinx</a>
              and a customised <a href="https://github.com/pradyunsg/furo">Furo</a> theme.
              Last updated on 2025-07-14T10:36:50+0200.
              This site will never display any ads: it is a non-profit project.
              It does not collect any data.
            </div>
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            In this section
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">References</a><ul>
<li><a class="reference internal" href="#how-to-cite">How to Cite</a></li>
<li><a class="reference internal" href="#bibliography">Bibliography</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="_static/documentation_options.js?v=937269b5"></script>
    <script src="_static/doctools.js?v=9bcbadda"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/scripts/furo.js?v=5fa4622c"></script>
    </body>
</html>