<!DOCTYPE html>


<html class="writer-html5" lang="en" >
<!-- Copyright (C) 2020-2022, Marek Gagolewski <https://www.gagolewski.com> -->

<head>
  <meta charset="utf-8" />
  <meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>References &mdash; Clustering Benchmarks</title>
  

  
  
  <script async defer src="https://buttons.github.io/buttons.js"></script>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/proof.css" type="text/css" />
  <link rel="stylesheet" href="_static/plot_directive.css" type="text/css" />
  <link rel="stylesheet" href="_static/graphviz.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/custom.css" type="text/css" />

  
  
    <link rel="shortcut icon" href="_static/favicon.png"/>
  

  
  

  
    <link rel="canonical" href="https://clustering-benchmarks.gagolewski.com/z_bibliography.html" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/proof.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Changelog" href="news.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html"> Clustering Benchmarks
          

          
          </a>

          <div class="version">
          by <a style="color: inherit" href="https://www.gagolewski.com">Marek Gagolewski</a><br />v1.1.1
          </div>

<!--
          
            
            
              <div class="version">
                by Marek Gagolewski
              </div>
            
          
-->

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search phrase..." />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">Methodology</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="index.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/true-vs-predicted.html">True vs Predicted Clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/noise-points.html">Noise Points</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/many-partitions.html">There Can Be Many Valid Partitions</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Benchmark Batteries</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="weave/suite-v1.html">Benchmark Suite (v1.1.0)</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/data-v1.html">Explore Datasets (v1.1.0)</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/results-v1.html">Clustering Results Repository (v1.1.0)</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/file-format.html">File Format Specification</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/how-to-access.html">Access from Python, R, MATLAB, etc.</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/contributing.html">Contributing</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Python API</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="weave/clustbench-usage.html">Using <em>clustbench</em></a></li>
<li class="toctree-l1"><a class="reference internal" href="clustbench-documentation.html">Documentation</a></li>
<li class="toctree-l1"><a class="reference external" href="https://github.com/gagolews/clustering-benchmarks">Source Code (GitHub)</a></li>
<li class="toctree-l1"><a class="reference external" href="https://pypi.org/project/clustering-benchmarks">PyPI Entry</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">See Also</span></p>
<ul>
<li class="toctree-l1"><a class="reference external" href="https://www.gagolewski.com/">Author's Homepage</a></li>
<li class="toctree-l1"><a class="reference external" href="https://datawranglingpy.gagolewski.com/">Data Wrangling in Python</a></li>
<li class="toctree-l1"><a class="reference external" href="https://genieclust.gagolewski.com/">genieclust Package</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Appendix</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="weave/colouriser.html">Colouriser: Planar Data Editor</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/external-validity-measures.html">External Cluster Validity Measures</a></li>
<li class="toctree-l1"><a class="reference internal" href="weave/internal-validity-measures.html">Side Note: Internal <em>(In)Validity</em> Measures</a></li>
<li class="toctree-l1"><a class="reference internal" href="news.html">Changelog</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">References</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#how-to-cite">How to Cite</a></li>
<li class="toctree-l2"><a class="reference internal" href="#bibliography">Bibliography</a></li>
</ul>
</li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Clustering Benchmarks</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content style-external-links">
        
          



















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>References</li>
    
    
      <li class="wy-breadcrumbs-aside">

        
        
        <a class="github-button" href="https://github.com/gagolews/clustering-benchmarks" data-icon="octicon-star" data-size="large" data-show-count="true" aria-label="Star gagolews/clustering-benchmarks on GitHub">GitHub</a>
        


        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
      
        <a href="news.html" class="btn btn-neutral float-left" title="Changelog" accesskey="p"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <section class="tex2jax_ignore mathjax_ignore" id="references">
<h1>References<a class="headerlink" href="#references" title="Permalink to this heading"></a></h1>
<section id="how-to-cite">
<h2>How to Cite<a class="headerlink" href="#how-to-cite" title="Permalink to this heading"></a></h2>
<p>When using this framework in research publications, please
cite <span id="id1">[<a class="reference internal" href="#id5" title="Gagolewski M. (2022).  A framework for benchmarking clustering algorithms. SoftwareX. in press. URL: https://clustering-benchmarks.gagolewski.com.">21</a>]</span> as specified below. Thank you.</p>
<p>Additionally, please mention the benchmark suite project
<span id="id2">[<a class="reference internal" href="#id6" title="Gagolewski M., et al. (2022).  A benchmark suite for clustering algorithms: version 1.1.0. URL: https://github.com/gagolews/clustering-data-v1/releases/tag/v1.1.0, DOI: 10.5281/zenodo.7088171.">26</a>]</span> as well as the literature references
listed in the description files corresponding to each dataset studied.</p>
</section>
<section id="bibliography">
<h2>Bibliography<a class="headerlink" href="#bibliography" title="Permalink to this heading"></a></h2>
<div class="docutils container" id="id3">
<dl class="citation">
<dt class="label" id="id49"><span class="brackets">1</span></dt>
<dd><p>Arbelaitz O., Gurrutxaga I., Muguerza J., Pérez J.M., Perona I. (2013).  An extensive comparative study of cluster validity indices. <em>Pattern Recognition</em>, 46(1):243–256. DOI: <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.patcog.2012.07.021">https://doi.org/10.1016/j.patcog.2012.07.021</a>.</p>
</dd>
<dt class="label" id="id40"><span class="brackets">2</span></dt>
<dd><p>Ball G.H., Hall D.J. (1965).  ISODATA: A novel method of data analysis and pattern classification. Technical Report AD699616, Stanford Research Institute.</p>
</dd>
<dt class="label" id="id73"><span class="brackets">3</span></dt>
<dd><p>Bezdek J.C., Keller J.M., Krishnapuram R., Kuncheva L.I., Pal N.R. (1999).  Will the real iris data please stand up? <em>IEEE Transactions on Fuzzy Systems</em>, 7(3):368–369. DOI: <a class="reference external" href="https://doi.org/10.1109/91.771092">10.1109/91.771092</a>.</p>
</dd>
<dt class="label" id="id50"><span class="brackets">4</span></dt>
<dd><p>Bezdek J.C., Pal N.R. (1998).  Some new indexes of cluster validity. <em>IEEE Transactions on Systems, Man, and Cybernetics, Part B (Cybernetics)</em>, 28(3):301–315. DOI: <a class="reference external" href="https://doi.org/10.1109/3477.678624">10.1109/3477.678624</a>.</p>
</dd>
<dt class="label" id="id74"><span class="brackets">5</span></dt>
<dd><p>Blum A., Hopcroft J., Kannan R. (2020).  <em>Foundations of Data Science</em>. Cambridge University Press. URL: <a class="reference external" href="https://www.cs.cornell.edu/jeh/book.pdf">https://www.cs.cornell.edu/jeh/book.pdf</a>.</p>
</dd>
<dt class="label" id="id22"><span class="brackets">6</span></dt>
<dd><p>Buitinck L., et al. (2013).  API design for machine learning software: Experiences from the scikit-learn project. In: <em>ECML PKDD Workshop: Languages for Data Mining and Machine Learning</em>, pp. 108–122.</p>
</dd>
<dt class="label" id="id37"><span class="brackets">7</span></dt>
<dd><p>Caliński T., Harabasz J. (1974).  A dendrite method for cluster analysis. <em>Communications in Statistics</em>, 3(1):1–27. DOI: <a class="reference external" href="https://doi.org/10.1080/03610927408827101">10.1080/03610927408827101</a>.</p>
</dd>
<dt class="label" id="id67"><span class="brackets">8</span></dt>
<dd><p>Chang H., Yeung D.Y. (2008).  Robust path-based spectral clustering. <em>Pattern Recognition</em>, 41(1):191–203.</p>
</dd>
<dt class="label" id="id15"><span class="brackets">9</span></dt>
<dd><p>Crouse D.F. (2016).  On implementing 2D rectangular assignment algorithms. <em>IEEE Transactions on Aerospace and Electronic Systems</em>, 52(4):1679–1696. DOI: <a class="reference external" href="https://doi.org/10.1109/TAES.2016.140952">10.1109/TAES.2016.140952</a>.</p>
</dd>
<dt class="label" id="id24"><span class="brackets">10</span></dt>
<dd><p>Dasgupta S., Ng V. (2009).  Single data, multiple clusterings. In: <em>Proc. NIPS Workshop Clustering: Science or Art? Towards Principled Approaches</em>. URL: <a class="reference external" href="https://clusteringtheory.org">https://clusteringtheory.org</a>.</p>
</dd>
<dt class="label" id="id38"><span class="brackets">11</span></dt>
<dd><p>Davies D.L., Bouldin D.W. (1979).  A cluster separation measure. <em>IEEE Transactions on Pattern Analysis and Machine Intelligence</em>, PAMI–1(2):224–227. DOI: <a class="reference external" href="https://doi.org/10.1109/TPAMI.1979.4766909">10.1109/TPAMI.1979.4766909</a>.</p>
</dd>
<dt class="label" id="id54"><span class="brackets">12</span></dt>
<dd><p>Dua D., Graff C. (2022).  UCI Machine Learning Repository. <a class="reference external" href="http://archive.ics.uci.edu/ml">http://archive.ics.uci.edu/ml</a>.</p>
</dd>
<dt class="label" id="id39"><span class="brackets">13</span></dt>
<dd><p>Dunn J.C. (1974).  A fuzzy relative of the ISODATA process and its use in detecting compact well-separated clusters. <em>Journal of Cybernetics</em>, 3(3):32–57. DOI: <a class="reference external" href="https://doi.org/10.1080/01969727308546046">10.1080/01969727308546046</a>.</p>
</dd>
<dt class="label" id="id44"><span class="brackets">14</span></dt>
<dd><p>Edwards A.W.F., Cavalli-Sforza L.L. (1965).  A method for cluster analysis. <em>Biometrics</em>, 21(2):362–375. DOI: <a class="reference external" href="https://doi.org/10.2307/2528096">10.2307/2528096</a>.</p>
</dd>
<dt class="label" id="id64"><span class="brackets">15</span></dt>
<dd><p>Fisher R.A. (1936).  The use of multiple measurements in taxonomic problems. <em>Annals of Eugenics</em>, 7:179–188.</p>
</dd>
<dt class="label" id="id58"><span class="brackets">16</span></dt>
<dd><p>Fowlkes E.B., Mallows C.L. (1983).  A method for comparing two hierarchical clusterings. <em>Journal of the American Statistical Association</em>, 78(383):553–569.</p>
</dd>
<dt class="label" id="id25"><span class="brackets">17</span></dt>
<dd><p>Fränti P., Sieranoja S. (2018).  K-means properties on six clustering benchmark datasets. <em>Applied Intelligence</em>, 48(12):4743–4759. DOI: <a class="reference external" href="https://doi.org/10.1007/s10489-018-1238-7">10.1007/s10489-018-1238-7</a>.</p>
</dd>
<dt class="label" id="id65"><span class="brackets">18</span></dt>
<dd><p>Fränti P., Virmajoki O. (2006).  Iterative shrinking method for clustering problems. <em>Pattern Recognition</em>, 39(5):761–765.</p>
</dd>
<dt class="label" id="id72"><span class="brackets">19</span></dt>
<dd><p>Fu L., Medico E. (2007).  FLAME: A novel fuzzy clustering method for the analysis of DNA microarray data. <em>BMC bioinformatics</em>, 8:3.</p>
</dd>
<dt class="label" id="id9"><span class="brackets">20</span></dt>
<dd><p>Gagolewski M. (2021).  genieclust: Fast and robust hierarchical clustering. <em>SoftwareX</em>, 15:100722. DOI: <a class="reference external" href="https://doi.org/10.1016/j.softx.2021.100722">10.1016/j.softx.2021.100722</a>.</p>
</dd>
<dt class="label" id="id5"><span class="brackets">21</span></dt>
<dd><p>Gagolewski M. (2022).  A framework for benchmarking clustering algorithms. <em>SoftwareX</em>. in press. URL: <a class="reference external" href="https://clustering-benchmarks.gagolewski.com">https://clustering-benchmarks.gagolewski.com</a>.</p>
</dd>
<dt class="label" id="id4"><span class="brackets">22</span></dt>
<dd><p>Gagolewski M. (2022).  Adjusted asymmetric accuracy: A well-behaving external cluster validity measure. under review (preprint). URL: <a class="reference external" href="https://arxiv.org/pdf/2209.02935.pdf">https://arxiv.org/pdf/2209.02935.pdf</a>, DOI: <a class="reference external" href="https://doi.org/10.48550/arXiv.2209.02935">10.48550/arXiv.2209.02935</a>.</p>
</dd>
<dt class="label" id="id7"><span class="brackets">23</span></dt>
<dd><p>Gagolewski M. (2022).  <em>Minimalist Data Wrangling with Python</em>. Zenodo, Melbourne. ISBN 978-0-6455719-1-2. URL: <a class="reference external" href="https://datawranglingpy.gagolewski.com/">https://datawranglingpy.gagolewski.com/</a>, DOI: <a class="reference external" href="https://doi.org/10.5281/zenodo.6451068">10.5281/zenodo.6451068</a>.</p>
</dd>
<dt class="label" id="id10"><span class="brackets">24</span></dt>
<dd><p>Gagolewski M., Bartoszuk M., Cena A. (2016).  Genie: A new, fast, and outlier-resistant hierarchical clustering algorithm. <em>Information Sciences</em>, 363:8–23. DOI: <a class="reference external" href="https://doi.org/10.1016/j.ins.2016.05.003">10.1016/j.ins.2016.05.003</a>.</p>
</dd>
<dt class="label" id="id8"><span class="brackets">25</span></dt>
<dd><p>Gagolewski M., Bartoszuk M., Cena A. (2021).  Are cluster validity measures (in)valid? <em>Information Sciences</em>, 581:620–636. DOI: <a class="reference external" href="https://doi.org/10.1016/j.ins.2021.10.004">10.1016/j.ins.2021.10.004</a>.</p>
</dd>
<dt class="label" id="id6"><span class="brackets">26</span></dt>
<dd><p>Gagolewski M., et al. (2022).  A benchmark suite for clustering algorithms: version 1.1.0. URL: <a class="reference external" href="https://github.com/gagolews/clustering-data-v1/releases/tag/v1.1.0">https://github.com/gagolews/clustering-data-v1/releases/tag/v1.1.0</a>, DOI: <a class="reference external" href="https://doi.org/10.5281/zenodo.7088171">10.5281/zenodo.7088171</a>.</p>
</dd>
<dt class="label" id="id70"><span class="brackets">27</span></dt>
<dd><p>Gionis A., Mannila H., Tsaparas P. (2007).  Clustering aggregation. <em>ACM Transactions on Knowledge Discovery from Data</em>, 1(1):4.</p>
</dd>
<dt class="label" id="id30"><span class="brackets">28</span></dt>
<dd><p>Graves D., Pedrycz W. (2010).  Kernel-based fuzzy clustering and fuzzy clustering: A comparative experimental study. <em>Fuzzy Sets and Systems</em>, 161:522–543. DOI: <a class="reference external" href="https://doi.org/10.1016/j.fss.2009.10.021">10.1016/j.fss.2009.10.021</a>.</p>
</dd>
<dt class="label" id="id36"><span class="brackets">29</span></dt>
<dd><p>Halkidi M., Batistakis Y., Vazirgiannis M. (2001).  On clustering validation techniques. <em>Journal of Intelligent Information Systems</em>, pages 107–145. DOI: <a class="reference external" href="https://doi.org/10.1023/A:1012801612483">10.1023/A:1012801612483</a>.</p>
</dd>
<dt class="label" id="id12"><span class="brackets">30</span></dt>
<dd><p>Hubert L., Arabie P. (1985).  Comparing partitions. <em>Journal of Classification</em>, 2(1):193–218. DOI: <a class="reference external" href="https://doi.org/10.1007/BF01908075">10.1007/BF01908075</a>.</p>
</dd>
<dt class="label" id="id71"><span class="brackets">31</span></dt>
<dd><p>Jain A., Law M. (2005).  Data clustering: A user's dilemma. <em>Lecture Notes in Computer Science</em>, 3776:1–10.</p>
</dd>
<dt class="label" id="id76"><span class="brackets">32</span></dt>
<dd><p>Jamil M., Yang X.-S., Zepernick H.-J. (2013).  8-test functions for global optimization: A comprehensive survey. In: <em>Swarm Intelligence and Bio-Inspired Computation</em>, pp. 193–222. DOI: <a class="reference external" href="https://doi.org/10.1016/B978-0-12-405163-8.00008-9">10.1016/B978-0-12-405163-8.00008-9</a>.</p>
</dd>
<dt class="label" id="id32"><span class="brackets">33</span></dt>
<dd><p>Karypis G., Han E.H., Kumar V. (1999).  CHAMELEON: Hierarchical clustering using dynamic modeling. <em>Computer</em>, 32(8):68–75. DOI: <a class="reference external" href="https://doi.org/10.1109/2.781637">10.1109/2.781637</a>.</p>
</dd>
<dt class="label" id="id17"><span class="brackets">34</span></dt>
<dd><p>Kvalseth T.O. (1987).  Entropy and correlation: Some comments. <em>IEEE Trans. Systems, Man and Cybernetics</em>, 17(3):517–519. DOI: <a class="reference external" href="https://doi.org/10.1109/TSMC.1987.4309069">10.1109/TSMC.1987.4309069</a>.</p>
</dd>
<dt class="label" id="id69"><span class="brackets">35</span></dt>
<dd><p>Kärkkäinen I., Fränti P. (2002).  Dynamic local search algorithm for the clustering problem. In: <em>Proc. 16th Intl. Conf. Pattern Recognition'02</em>, volume 2, pp. 240–243. IEEE.</p>
</dd>
<dt class="label" id="id47"><span class="brackets">36</span></dt>
<dd><p>Lloyd S.P. (1957).  Least squares quantization in PCM. <em>IEEE Transactions on Information Theory</em>, 28:128–137. Originally a 1957 Bell Telephone Laboratories Research Report; republished in 1982. DOI: <a class="reference external" href="https://doi.org/10.1109/TIT.1982.1056489">10.1109/TIT.1982.1056489</a>.</p>
</dd>
<dt class="label" id="id35"><span class="brackets">37</span></dt>
<dd><p>Maulik U., Bandyopadhyay S. (2002).  Performance evaluation of some clustering algorithms and validity indices. <em>IEEE Transactions on Pattern Analysis and Machine Intelligence</em>, 24(12):1650–1654. DOI: <a class="reference external" href="https://doi.org/10.1109/TPAMI.2002.1114856">10.1109/TPAMI.2002.1114856</a>.</p>
</dd>
<dt class="label" id="id33"><span class="brackets">38</span></dt>
<dd><p>McInnes L., Healy J., Astels S. (2017).  hdbscan: Hierarchical density based clustering. <em>The Journal of Open Source Software</em>, 2(11):205. DOI: <a class="reference external" href="https://doi.org/10.21105/joss.00205">10.21105/joss.00205</a>.</p>
</dd>
<dt class="label" id="id55"><span class="brackets">39</span></dt>
<dd><p>Meilă M., Heckerman D. (2001).  An experimental comparison of model-based clustering methods. <em>Machine Learning</em>, 42:9–29. DOI: <a class="reference external" href="https://doi.org/10.1023/A:1007648401407">10.1023/A:1007648401407</a>.</p>
</dd>
<dt class="label" id="id34"><span class="brackets">40</span></dt>
<dd><p>Milligan G.W., Cooper M.C. (1985).  An examination of procedures for determining the number of clusters in a data set. <em>Psychometrika</em>, 50(2):159–179.</p>
</dd>
<dt class="label" id="id75"><span class="brackets">41</span></dt>
<dd><p>Müller A.C., Nowozin S., Lampert C.H. (2012).  Information theoretic clustering using minimum spanning trees. In: <em>Proc. German Conference on Pattern Recognition</em>. URL: <a class="reference external" href="https://github.com/amueller/information-theoretic-mst">https://github.com/amueller/information-theoretic-mst</a>.</p>
</dd>
<dt class="label" id="id13"><span class="brackets">42</span></dt>
<dd><p>Müllner D. (2013).  fastcluster: Fast hierarchical, agglomerative clustering routines for R and Python. <em>Journal of Statistical Software</em>, 53(9):1–18. DOI: <a class="reference external" href="https://doi.org/10.18637/jss.v053.i09">10.18637/jss.v053.i09</a>.</p>
</dd>
<dt class="label" id="id23"><span class="brackets">43</span></dt>
<dd><p>Pedregosa F., et al. (2011).  Scikit-learn: Machine learning in Python. <em>Journal of Machine Learning Research</em>, 12(85):2825–2830. URL: <a class="reference external" href="http://jmlr.org/papers/v12/pedregosa11a.html">http://jmlr.org/papers/v12/pedregosa11a.html</a>.</p>
</dd>
<dt class="label" id="id59"><span class="brackets">44</span></dt>
<dd><p>Rand W.M. (1971).  Objective criteria for the evaluation of clustering methods. <em>Journal of the American Statistical Association</em>, 66(336):846–850. DOI: <a class="reference external" href="https://doi.org/10.2307/2284239">10.2307/2284239</a>.</p>
</dd>
<dt class="label" id="id16"><span class="brackets">45</span></dt>
<dd><p>Rezaei M., Fränti P. (2016).  Set matching measures for external cluster validity. <em>IEEE Transactions on Knowledge and Data Engineering</em>, 28(8):2173–2186. DOI: <a class="reference external" href="https://doi.org/10.1109/TKDE.2016.2551240">10.1109/TKDE.2016.2551240</a>.</p>
</dd>
<dt class="label" id="id41"><span class="brackets">46</span></dt>
<dd><p>Rousseeuw P.J. (1987).  Silhouettes: A graphical aid to the interpretation and validation of cluster analysis. <em>Journal of Computational and Applied Mathematics</em>, 20:53–65. DOI: <a class="reference external" href="https://doi.org/10.1016/0377-0427(87)90125-7">10.1016/0377-0427(87)90125-7</a>.</p>
</dd>
<dt class="label" id="id62"><span class="brackets">47</span></dt>
<dd><p>Sieranoja S., Fränti P. (2019).  Fast and general density peaks clustering. <em>Pattern Recognition Letters</em>, 128:551–558. DOI: <a class="reference external" href="https://doi.org/10.1016/j.patrec.2019.10.019">10.1016/j.patrec.2019.10.019</a>.</p>
</dd>
<dt class="label" id="id56"><span class="brackets">48</span></dt>
<dd><p>Steinley D. (2004).  Properties of the Hubert–Arabie adjusted Rand index. <em>Psychological Methods</em>, 9(3):386–396. DOI: <a class="reference external" href="https://doi.org/10.1037/1082-989X.9.3.386">10.1037/1082-989X.9.3.386</a>.</p>
</dd>
<dt class="label" id="id28"><span class="brackets">49</span></dt>
<dd><p>Thrun M.C., Ultsch A. (2020).  Clustering benchmark datasets exploiting the fundamental clustering problems. <em>Data in Brief</em>, 30:105501. DOI: <a class="reference external" href="https://doi.org/10.1016/j.dib.2020.105501">10.1016/j.dib.2020.105501</a>.</p>
</dd>
<dt class="label" id="id82"><span class="brackets">50</span></dt>
<dd><p>Ullmann T., Beer A., Hünemörder M., Seidl T., Boulesteix A.-L. (2022).  Over-optimistic evaluation and reporting of novel cluster algorithms: An illustrative study. <em>Advances in Data Analysis and Classification</em>. DOI: <a class="reference external" href="https://doi.org/10.1007/s11634-022-00496-5">10.1007/s11634-022-00496-5</a>.</p>
</dd>
<dt class="label" id="id84"><span class="brackets">51</span></dt>
<dd><p>Ullmann T., Hennig C., Boulesteix A.-L. (2021).  Validation of cluster analysis results on validation data: A systematic framework. <em>Wiley Interdisciplinary Reviews: Data Mining and Knowledge Discovery</em>, 12(3):e14444. DOI: <a class="reference external" href="https://doi.org/10.1002/widm.1444">10.1002/widm.1444</a>.</p>
</dd>
<dt class="label" id="id27"><span class="brackets">52</span></dt>
<dd><p>Ultsch A. (2005).  Clustering with SOM: U*C. In: <em>Workshop on Self-Organizing Maps</em>, pp. 75–82.</p>
</dd>
<dt class="label" id="id68"><span class="brackets">53</span></dt>
<dd><p>Veenman C.J., Reinders M.J.T., Backer E. (2002).  A maximum variance cluster algorithm. <em>IEEE Transactions on Pattern Analysis and Machine Intelligence</em>, 24(9):1273–1280.</p>
</dd>
<dt class="label" id="id18"><span class="brackets">54</span></dt>
<dd><p>Vinh N.X., Epps J., Bailey J. (2010).  Information theoretic measures for clusterings comparison: Variants, properties, normalization and correction for chance. <em>Journal of Machine Learning Research</em>, 11(95):2837–2854. URL: <a class="reference external" href="http://jmlr.org/papers/v11/vinh10a.html">http://jmlr.org/papers/v11/vinh10a.html</a>.</p>
</dd>
<dt class="label" id="id51"><span class="brackets">55</span></dt>
<dd><p>von Luxburg U., Williamson R.C., Guyon I. (2012).  Clustering: science or art? In: Guyon I., et al., editors, <em>Proc. ICML Workshop on Unsupervised and Transfer Learning</em>, volume 27 of Proc. Machine Learning Research, pp. 65–79.</p>
</dd>
<dt class="label" id="id45"><span class="brackets">56</span></dt>
<dd><p>Ward Jr. J.H. (1963).  Hierarchical grouping to optimize an objective function. <em>Journal of the American Statistical Association</em>, 58(301):236–244. DOI: <a class="reference external" href="https://doi.org/10.1080/01621459.1963.10500845">10.1080/01621459.1963.10500845</a>.</p>
</dd>
<dt class="label" id="id77"><span class="brackets">57</span></dt>
<dd><p>Weise T., et al. (2014).  Benchmarking optimization algorithms: An open source framework for the traveling salesman problem. <em>IEEE Computational Intelligence Magazine</em>, 9(3):40–52. DOI: <a class="reference external" href="https://doi.org/10.1109/MCI.2014.2326101">10.1109/MCI.2014.2326101</a>.</p>
</dd>
<dt class="label" id="id61"><span class="brackets">58</span></dt>
<dd><p>Xiao H., Rasul K., Vollgraf R. (2017).  Fashion-MNIST: A novel image dataset for benchmarking machine learning algorithms. URL: <a class="reference external" href="https://arxiv.org/pdf/1708.07747.pdf">https://arxiv.org/pdf/1708.07747.pdf</a>, DOI: <a class="reference external" href="https://doi.org/10.48550/arXiv.1708.07747">10.48550/arXiv.1708.07747</a>.</p>
</dd>
<dt class="label" id="id48"><span class="brackets">59</span></dt>
<dd><p>Xu Q., Zhang Q., Liu J., Luo B. (2020).  Efficient synthetical clustering validity indexes for hierarchical clustering. <em>Expert Systems with Applications</em>, 151:113367. DOI: <a class="reference external" href="https://doi.org/10.1016/j.eswa.2020.113367">10.1016/j.eswa.2020.113367</a>.</p>
</dd>
<dt class="label" id="id66"><span class="brackets">60</span></dt>
<dd><p>Zahn C.T. (1971).  Graph-theoretical methods for detecting and describing gestalt clusters. <em>IEEE Transactions on Computers</em>, C-20(1):68–86.</p>
</dd>
<dt class="label" id="id63"><span class="brackets">61</span></dt>
<dd><p>Zhang T., Ramakrishnan R., Livny M. (1997).  BIRCH: A new data clustering algorithm and its applications. <em>Data Mining and Knowledge Discovery</em>, 1:141–182. DOI: <a class="reference external" href="https://doi.org/10.1023/A:1009783824328">10.1023/A:1009783824328</a>.</p>
</dd>
<dt class="label" id="id83"><span class="brackets">62</span></dt>
<dd><p>Van Mechelen I., et al. (2018).  Benchmarking in cluster analysis: A white paper. URL: <a class="reference external" href="https://arxiv.org/pdf/1809.10496.pdf">https://arxiv.org/pdf/1809.10496.pdf</a>.</p>
</dd>
</dl>
</div>
</section>
</section>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="news.html" class="btn btn-neutral float-left" title="Changelog" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        Copyright &#169; 2020–2022 by <a href="https://www.gagolewski.com">Marek Gagolewski</a>. Some rights reserved. Licensed under <a href='https://creativecommons.org/licenses/by-nc-nd/4.0/'>CC BY-NC-ND 4.0</a>.

    Built with <a href="https://sphinx-doc.org/">Sphinx</a>
    and a customised <a href="https://github.com/rtfd/sphinx_rtd_theme">rtd</a>
    theme.
      <span class="lastupdated">
        Last updated on 2022-11-16T14:17:27+1100.
      </span>


    This site will never display any ads: it is a non-profit project.
    It does not collect any data.

    </p>
  </div> 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>